pom.xml

```diff
@@ -0,0 +1,152 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+   <modelVersion>4.0.0</modelVersion>
+   <groupId>com.pingcap.tikv</groupId>
+   <artifactId>kvproto</artifactId>
+   <version>0.1.0</version>
+   <packaging>jar</packaging>
+   <name>kvproto</name>
+   <description>Protobuf definitions for TiKV</description>
+   <organization>
+      <name>PingCAP</name>
+   </organization>
+   <properties>
+      <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+      <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+      <protobuf.version>3.1.0</protobuf.version>
+      <grpc.version>1.4.0</grpc.version>
+   </properties>
+   <dependencies>
+      <!-- grpc dependencies -->
+      <dependency>
+         <groupId>io.grpc</groupId>
+         <artifactId>grpc-netty</artifactId>
+         <version>${grpc.version}</version>
+      </dependency>
+      <dependency>
+         <groupId>io.grpc</groupId>
+         <artifactId>grpc-protobuf</artifactId>
+         <version>${grpc.version}</version>
+      </dependency>
+      <dependency>
+         <groupId>io.grpc</groupId>
+         <artifactId>grpc-stub</artifactId>
+         <version>${grpc.version}</version>
+      </dependency>
+   </dependencies>
+   <build>
+      <extensions>
+         <extension>
+            <groupId>kr.motd.maven</groupId>
+            <artifactId>os-maven-plugin</artifactId>
+            <version>1.4.1.Final</version>
+         </extension>
+      </extensions>
+      <plugins>
+         <plugin>
+            <artifactId>maven-resources-plugin</artifactId>
+            <version>2.5</version>
+            <executions>
+               <execution>
+                  <id>copy-resources</id>
+                  <phase>validate</phase>
+                  <goals>
+                     <goal>copy-resources</goal>
+                  </goals>
+                  <configuration>
+                     <outputDirectory>${basedir}/proto/gogoproto</outputDirectory>
+                     <resources>
+                        <resource>
+                           <directory>${GOPATH}/src/github.com/gogo/protobuf/gogoproto</directory>
```

Comment:
this is equal to generate-rust.sh
gogoproto introduce special tags and options into proto files and this has to be compiled with   gogoproto dependencies.